src/custom-elements/src/components/popover/popover.tsx

```diff
@@ -0,0 +1,372 @@
+        this._preAnimationTransformStyle = `translate3d(${
+          data.popper.left
+        }px, ${data.popper.top - 20}px, 0)`;
+      } else if (data.placement.indexOf('left') === 0) {
+        this._preAnimationTransformStyle = `translate3d(${data.popper.left +
+          20}px, ${data.popper.top}px, 0)`;
+      } else {
+        this._preAnimationTransformStyle = data.styles.transform;
+      }
+      return data;
+    };
+
+    this._popper = new Popper(this._reference, this._popoverElement, {
+      modifiers: {
+        applyStyle: { enabled: true },
+        applyChiStyle: {
+          enabled: true,
+          fn: savePopperData,
+          // Set order to run after popper applyStyle modifier
+          // as we need data.styles to be filled.
+          order: 875
+        },
+        arrow: {
+          element: '.a-arrow',
+          enabled: this.arrow
+        }
+      },
+      removeOnDestroy: false,
+      placement: this.position
+    });
+
+    this._popper.update();
+  }
+
+  private _destroyPopper() {
+    this._preAnimationTransformStyle = null;
+    this._postAnimationTransformStyle = null;
+    if (this._popper) {
+      this._popper.destroy();
+    }
+  }
+
+  componentWillLoad(): void {
+    this.positionValidation(this.position);
+    this.referenceElementChanged(this.reference);
+    if (this.active) {
+      this._animationClasses = CLASSES.ACTIVE;
+    }
+  }
+
+  componentDidLoad(): void {
+    this._documentClickHandler = () => {
+      !this._closePrevented && !this.preventAutoHide && this.hide();
+    };
+    this._documentKeyHandler = e => {
+      if (
+        !this.preventAutoHide &&
+        'key' in e &&
+        (e.key === 'Escape' || e.key === 'Esc' || e.key === ESCAPE_KEYCODE)
+      ) {
+        this.hide();
+      }
+    };
+    this._configurePopoverPopper();
+    this._componentLoaded = true;
+    document.addEventListener('click', this._documentClickHandler);
+    document.addEventListener('keyup', this._documentKeyHandler);
+  }
+
+  componentDidUnload(): void {
+    this._destroyPopper();
+    if (this.currentAnimation && !this.currentAnimation.isStopped()) {
+      this.currentAnimation.stop();
+    }
+    this.currentAnimation = null;
+    this._componentLoaded = false;
+    document.removeEventListener('click', this._documentClickHandler);
+    document.removeEventListener('keyup', this._documentKeyHandler);
+  }
+
+  componentDidUpdate(): void {
+    while (this._didUpdateCallBackOnceQueue.length) {
+      const cb = this._didUpdateCallBackOnceQueue.shift();
+      cb();
+    }
+  }
+
+  preventAutoClose() {
+    this._closePrevented = true;
+    window.clearTimeout(this._closePreventedTimeout);
+    this._closePreventedTimeout = window.setTimeout(() => {
+      this._closePrevented = false;
+    }, 100);
+  }
+
+  hostData() {
+    return {
+      'aria-hidden': this.active ? 'false' : 'true'
+    };
+  }
+
+  render() {
+    return (
+      <div
+        class={`m-popover
```

Comment:
@dani-cl-madrid shadow should be included in this level instead of the inner html tag as shadow is not rendering well on the arrow.